wallet/src/actors/worker/methods.rs

```diff
@@ -313,53 +319,92 @@ impl Worker {
-    pub fn get_input_values_from_txns(
+    /// Extend transactions with metadata requested to the node through JSON-RPC queries.
+    pub fn extend_transactions_data(
         &self,
-        txns: &[types::Transaction],
-    ) -> Result<Vec<Vec<ValueTransferOutput>>> {
-        let txns_output_pointers: Vec<Vec<&types::OutputPointer>> = txns
+        txns: Vec<types::Transaction>,
+    ) -> Result<Vec<model::ExtendedTransaction>> {
+        let queries: Vec<Option<IndexTransactionQuery>> = txns
             .iter()
             .map(|txn| match txn {
-                types::Transaction::ValueTransfer(vt) => vt
-                    .body
-                    .inputs
-                    .iter()
-                    .map(|input| input.output_pointer())
-                    .collect(),
-                types::Transaction::DataRequest(dr) => dr
-                    .body
-                    .inputs
-                    .iter()
-                    .map(|input| input.output_pointer())
-                    .collect(),
-                types::Transaction::Commit(commit) => commit
-                    .body
-                    .collateral
-                    .iter()
-                    .map(|input| input.output_pointer())
-                    .collect(),
-                _ => vec![],
+                types::Transaction::ValueTransfer(vt) => {
+                    Some(IndexTransactionQuery::InputTransactions(
+                        vt.body
+                            .inputs
+                            .iter()
+                            .map(|input| input.output_pointer().clone())
+                            .collect(),
+                    ))
+                }
+                types::Transaction::DataRequest(dr) => {
+                    Some(IndexTransactionQuery::InputTransactions(
+                        dr.body
+                            .inputs
+                            .iter()
+                            .map(|input| input.output_pointer().clone())
+                            .collect(),
+                    ))
+                }
+                types::Transaction::Commit(commit) => {
+                    Some(IndexTransactionQuery::InputTransactions(
+                        commit
+                            .body
+                            .collateral
+                            .iter()
+                            .map(|input| input.output_pointer().clone())
+                            .collect(),
+                    ))
+                }
+                types::Transaction::Tally(tally) => Some(IndexTransactionQuery::DataRequestReport(
+                    tally.dr_pointer.to_string(),
+                )),
+                _ => None,
             })
             .collect();
 
-        let query_vtt_outputs_futures: Result<Vec<Vec<ValueTransferOutput>>> = txns_output_pointers
+        let query_results: Result<Vec<model::ExtendedTransaction>> = txns
             .iter()
```

Comment:
You should be able to remove all the `transaction.clone()` from below if you make this iterator take `txns` by value:

```suggestion
            .into_iter()
```